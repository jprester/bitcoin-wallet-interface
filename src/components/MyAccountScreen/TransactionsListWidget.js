import React from 'react';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

import TransactionList from './TransactionList';


const styles = theme => ({
  root: {
    width: '100%',
    maxWidth: 360,
    marginTop: 50,
    backgroundColor: theme.palette.background.paper,
  },

  transactionAdress: {
    fontSize: 11,
  },

  transactionAmount: {
    fontSize: 11,
  },
});

export const TransactionsListWidget = (props, classes) => (
  <div className={classes.root}>
    <h2> Transactions </h2>
    <TransactionList classes={classes} {...props} />
  </div>
);

const mapStateToProps = state => ({
  transactionList: state.transactionsData.transactions,
});

const styledComponent = withStyles(styles)(TransactionsListWidget);

export default connect(mapStateToProps)(styledComponent);

TransactionsListWidget.propTypes = {
  classes: PropTypes.shape({}).isRequired,
};
