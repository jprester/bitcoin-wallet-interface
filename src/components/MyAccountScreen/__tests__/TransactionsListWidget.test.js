import React from 'react';
import { shallow } from 'enzyme';
import { TransactionsListWidget } from '../TransactionsListWidget';

const wrapper = shallow((<TransactionsListWidget classes={{}} transactionList={[{ name: 'name', email: 'email@email.com', amount: 9 }]} />));

describe('TransactionsListWidget', () => {
  it('should render TransactionsListWidget correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
