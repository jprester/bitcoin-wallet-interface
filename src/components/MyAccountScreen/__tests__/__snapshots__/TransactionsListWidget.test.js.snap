// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsListWidget should render TransactionsListWidget correctly 1`] = `
<div>
  <h2>
     Transactions 
  </h2>
  <TransactionList
    classes={Object {}}
    transactionList={
      Array [
        Object {
          "amount": 9,
          "email": "email@email.com",
          "name": "name",
        },
      ]
    }
  />
</div>
`;
